### PR TITLE
ts/indicator-price-envelopes

### LINKED ISSUE #11854: Migrate `indicators/price-envelopes` to TypeScript

### PR DESCRIPTION
**Pull request by activist**
Closes #11854, Tools: Migrated PriceEnvelopesIndicator to TS.

### CODE REVIEW

**Comment by activist** [comment]
Awaiting merge of #12124 for further work.

**Comment by bre1470** [comment]
There is no change in `.eslintignore`.

**Comment by activist** [comment]
> There is no change in `.eslintignore`.

Thanks, surprisingly the commit didn't produce any errors after being converted to ts. Any idea why? Too me it looks like the output js should fail on e.g. max line length.

**Comment by bre1470** [comment]
I guess, somehow every case is matching the excluding pattern: `(eslint-disable-line|pointFormat|headerFormat| highcharts\/|\\]\\(#|@apioption|@borrows|@default|@extends|@name|@param|@property|@return|@sample|@see|@type|@typedef|@validvalue)` 🤷‍♀️

File: ts/indicators/price-envelopes.src.ts [hunk 1]
```diff
@@ -0,0 +1,339 @@
...
+            public points: Array<PriceEnvelopesIndicatorPoint>;
+            public pointValKey: string;
+            public toYData(
+                point: PriceEnvelopesIndicatorPoint
+            ): [number, number, number];
+            public translate(): void;
+        }
+
+        interface PriceEnvelopesIndicatorGappedExtensionObject {
+            [index: string]: Dictionary<(number|undefined)>;
```
Comment (bre1470) [inline ts/indicators/price-envelopes.src.ts]: If you go that far, you can remove the interface and use `Dictionary<Dictionary<(number|undefined)>>`.

File: ts/indicators/price-envelopes.src.ts [hunk 2]
```diff
@@ -0,0 +1,339 @@
...
+        }
+
+        interface PriceEnvelopesIndicatorOptionsParams
```
Comment (bre1470) [inline ts/indicators/price-envelopes.src.ts]: Has to be named `PriceEnvelopesIndicatorParamsOptions`.

File: ts/indicators/price-envelopes.src.ts (hunk 1 above)
Comment (activist) [reply ts/indicators/price-envelopes.src.ts]: This same structure seems to appear in a few different indicators. Could it be better to define this as separate interfaces in one of the the indicators and the reuse the same in the other indicators, or do you think that they should be declared a separate interfaces per indicator?

Comment (bre1470) [reply ts/indicators/price-envelopes.src.ts]: That is hard to tell, but as there is no mixin in use, I would go with separate interfaces.

Comment (activist) [reply ts/indicators/price-envelopes.src.ts]: Added separate interfaces. I hope the naming is ok `PriceEnvelopesIndicatorGappedExtensionObject` and `PriceEnvelopesIndicatorGappedExtensionOptions`

File: ts/indicators/price-envelopes.src.ts (hunk 2 above)
Comment (activist) [reply ts/indicators/price-envelopes.src.ts]: Fixed

**Pull request merged**